### PR TITLE
Timeout `du` after a minute.

### PR DESCRIPTION
**Pull request by vishh**
cc @timstclair

### CODE REVIEW

**Comment by vishh** [comment]
@k8s-bot test this

File: container/docker/fsHandler.go [hunk 1]
```diff
@@ -66,12 +69,12 @@ func (fh *realFsHandler) needsUpdate() bool {
...
 func (fh *realFsHandler) update() error {
 	// TODO(vishh): Add support for external mounts.
-	baseUsage, err := fh.fsInfo.GetDirUsage(fh.rootfs)
+	baseUsage, err := fh.fsInfo.GetDirUsage(fh.rootfs, duTimeout)
 	if err != nil {
 		return err
 	}
 
-	extraDirUsage, err := fh.fsInfo.GetDirUsage(fh.extraDir)
+	extraDirUsage, err := fh.fsInfo.GetDirUsage(fh.extraDir, duTimeout)
```
Comment (timstclair) [inline container/docker/fsHandler.go]: I think we should log these errors at `Error` level (here: https://github.com/vishh/cadvisor/blob/timeout-du/container/docker/fsHandler.go#L99 - I can't figure out how to comment there on GH)

Comment (vishh) [inline container/docker/fsHandler.go]: done

**Comment by timstclair** [comment]
LGTM, but I'm concerned that if `du` times out, it may indicate a recurring problem (i.e. rerunning du may also timeout). If this seems to be the case, we should consider marking it as a permanent failure (maybe give it a few tries).

**Comment by vishh** [comment]
How do we differentiate between transient and permanent failures? A disk
could become slow temporarily.

**Comment by timstclair** [comment]
Maybe an exponential backoff on the retries would be better.

**Comment by vishh** [comment]
@timstclair: I added exponential backoff. Limited it to a factor of `20` for now.

**Comment by vishh** [comment]
@timstclair: I fixed #501 as part of this PR and re-enabled the disk e2e. PTAL.

**Comment by vishh** [comment]
@timstclair: Addressed comments.

**Comment by timstclair** [comment]
Commented on https://github.com/vishh/cadvisor/commit/a5decfeaad26fcf72ce7254e4516ef18b21eacb5

**Comment by vishh** [comment]
@timstclair: This PR is ready to be reviewed again.

File: integration/tests/api/docker_test.go [hunk 2]
```diff
@@ -292,36 +292,60 @@ func TestDockerContainerNetworkStats(t *testing.T) {
...
+	switch storageDriver {
+	case framework.Aufs, framework.Overlay:
+		needsBaseUsageCheck = true
+	}
+	pass := false
+	// We need to wait for the `dd` operation to complete.
+	for i := 0; i < 10; i++ {
+		containerInfo, err := fm.Cadvisor().ClientV2().Stats(containerId, request)
+		require.NoError(t, err)
+		require.True(t, len(containerInfo) == 1)
```
Comment (timstclair) [inline integration/tests/api/docker_test.go]: (nit) prefer `require.Equal(t, 1, len(containerInfo))`

**Comment by timstclair** [comment]
LGTM

**Comment by a86c6f7964** [comment]
Any idea when this will be released, and/or if there is a good work around for the mean time?

**Comment by vishh** [comment]
We will try to make a release by the end of this week.

File: integration/tests/api/docker_test.go (hunk 2 above)
Comment (vishh) [inline integration/tests/api/docker_test.go]: Fixing

**Comment by vishh** [comment]
@timstclair: Addressed the two pending comments. This PR is ready to be merged now.

**Comment by timstclair** [comment]
LGTM, ok to self merge once e2e passes.

**Pull request merged**